---
title: "Projects"
bibliography: TCGA.bib
csl: styles.ref/genomebiology.csl
author: "Mikhail Dozmorov"
date: "`r Sys.Date()`"
output: 
  html_document:
    toc: true
    # toc_float: true
    theme: united
    # theme: cerulean
    # number_sections: true
  word_document:
    reference_docx: styles.doc/NIH_grant_style.docx
---

```{r setup, echo=FALSE, message=FALSE, warning=FALSE}
# Set up the environment
library(knitr)
opts_chunk$set(cache.path='cache/', fig.path='img/', cache=F, tidy=T, fig.keep='high', echo=F, dpi=100, warnings=F, message=F, comment=NA, warning=F, results='as.is', fig.width = 10, fig.height = 6) #out.width=700, 
library(pander)
panderOptions('table.split.table', Inf)
set.seed(1)
library(dplyr)
options(stringsAsFactors = FALSE)
```


# Project Title:    **"Developing a methylation-based survival analysis"**

### Provide a brief synopsis of the project:

Efforts such as The Cancer Genome Atlas (TCGA) collected massive -omics datasets. Gene expression has been extensively used in survival analysis, with the goal to identify one or several genes with expression predictive of survival. Methylation data has been largely overlooked, partly because of its size and complexity - one gene has several methylation sites with different survival prediction ability. Furthermore, genes have different length, complicating summary of methylation signal per gene. The goal of this project is to develop a method for survival analysis utilizing methylation data.

This project involves:

- Working with methylation and gene expression data from TCGA
- Performing survival analysis using methylation signal from individual CpG sites
- Accounting for distance between methylation sites and the start of a gene
- Developing a novel statistical model to summarize methylation signal within a gene, normalizing the signal per-unit-length of a gene
- Comparing the performance of methylation and gene expression data, or a combination of both, in survival analysis

### What outcomes/deliverables will the student produce for the Biostatistics research day?

Overview of survival analysis. Presenting the performance of a newly developed method of summarizing methylation signal per gene and using it for survival analysis. 

### Current status and ToDo

- Glioblastoma CpG sites predictive of survival using Cox PH model. DONE

- Annotating CpGs with overlap with (categorical) and proximity to (continuous) genomic features  
  - Using the original Illumina 450K annotations, from `FDb.InfiniumMethylation.hg19` package (??? which features). DONE  
  - Gene-centric overlap/proximity annotation. ToDo.  
    - Check http://www.haowulab.org/teaching/bioc/genomicRanges.pdf, and vignettes for the corresponding packages, this is the right way of finding relationships among genomic regions.
    - Use hg19 genomic coordinates from https://github.com/stephenturner/annotables. `grch37` object.
    - For overlap annotation (Yes/No), use the fact of CpG site being located within a gene.
    - For proximity annotation (shortest distance to TSS), note strand. Transcription Start Sites (TSS) start at the leftmost coordinate for genes on positive strant, and on rightmost coordinate for genes on negative strand
    - For proxomity annotation, if a CpG site is downstream/upstream of a TSS - use negative/positive sign for the distance

- Predictive model that can predict CpG sites associated with survival vs. non-associated
  - Use logistic regression. ToDo
  - There are many more non-associated CpG sites. Explore SMOTE package for accounting for such disbalance. https://github.com/amunategui/SMOTE-Oversample-Rare-Events. ToDo

- Add additional annotations for CpG sites  

  - Use GenomeRunner data, Merlot
  
  - /home/sequencing/data/ExtData/db_5.00_07-22-2015/custom_data/gfs/hg19
    - 3D_subcompartments - use overlap annotation only. Two types of overlap:
      - A-B - two-level annotation. Whether CpG sites overlap GSE63525_GM12878_subcompartments_A.bed or GSE63525_GM12878_subcompartments_B.bed. Are there CpG sites that overlap none?
      -A12-B1234 - four-level annotation. Whether CpG sites overlap A1/A2/B1/B2/B3/B4
    - DGV - use overlap annotation only. Each track there is its own annotation with two levels - Yes/No - a CpG overlaps or not with it. See http://dgv.tcag.ca/v104_20141022/app/faq?ref=#q12
    - GERP - use overlap annotation, Yes/No. Also, the file contains "score" - 5th column. Use it for continuous annotation, a CpG overlapping a GERP region is annotated with that numerical score, non-overlapping - 0.
    - nestedRepeats - use overlap annotation, Yes/No. Each track there is its own annotation with two levels - Yes/No - a CpG overlaps or not with it. See https://en.wikipedia.org/wiki/Repeated_sequence_(DNA)#Main_types
    - super_enhancers - use overlap annotation, Yes/No. Each track there is its own annotation with two levels - Yes/No - a CpG overlaps or not with it. We need Gm12878 and K562.
    - UCNE - use overlap annotation, Yes/No. Each track there is its own annotation with two levels - Yes/No - a CpG overlaps or not with it. See http://ccg.vital-it.ch/UCNEbase/
    - VMR - use overlap annotation, Yes/No. Each track there is its own annotation with two levels - Yes/No - a CpG overlaps or not with it. These are Variably Methylated Regions

  - /home/sequencing/data/ExtData/db_5.00_07-22-2015/grsnp_db/hg19/ENCODE/chromStates
    - Use `BroadHmm` and `Combined` folders. `Gm12878` and `K562` subfolders. 
    - Use overlap annotation, Yes/No. Each track there is its own annotation with two levels - Yes/No - a CpG overlaps or not with it. See https://github.com/mdozmorov/genomerunner_web/wiki/ENCODE-chromStates

### Research plan

- Data folder on Merlot; /home/sequencing/work/Methylation/TCGA_BRCA_methylation
    - It contains methylation data for all BRCA samples, and 4 scripts to split it into 4 datasets corresponding to different types of BRCA
    - The data format is: CpG sites (rows) X samples (columns)
    - The datasets are large, ~450K rows

- Samples have clinical annotations that include overall survival and/or disease-free survival and other covariates. See https://www.ncbi.nlm.nih.gov/pubmed/26704973 and other packages referenced therein how to obtain clinical data

- **Aim 1:** Define the role of location of methylation sites on survival.
    - Process each BRCA type separately.
    - Find CpGs with methylation level associated with survival.  See `survival` package. Use Cox PH model to account for other covariates. 
    - Try $\beta$-values and $M$-values.
    - Given the location of each CpG site, both continuous (length, in bp) and categorical (promoter, 1st exon, etc.), find whether certain locations are more significantly associated with survival than can be expected by chance.

- **Aim 2:** Define the role of location of a set of CpG sites with methylation level distinguishing four BRCA types.
    - The goal is to define a classifier (a set of CpG sites) predicting BRCA type.
    - Use random forest algorithm, and other classifiers in the `caret` package.
    - Use the same location approach to find whether classifier CpGs have location preference.

### References 

[@anaya2016oncolnc; @anaya2016oncorank; @anaya2016pan; @Colaprico:2016aa; @diez2013survival; @goel2010understanding; @Gyorffy:2015aa; @Ring:2016aa; @silva2016tcga]