---
title: "TCGA survival analysis"
author: "Mikhail Dozmorov"
date: "`r Sys.Date()`"
output:
  pdf_document:
    toc: no
  html_document:
    theme: united
    toc: yes
csl: styles.ref/genomebiology.csl
bibliography: data.TCGA/TCGA.bib
---

```{r setup, echo=FALSE, message=FALSE, warning=FALSE}
# Set up the environment
library(knitr)
opts_chunk$set(cache.path='cache/', fig.path='img/', cache=F, tidy=T, fig.keep='high', echo=F, dpi=100, warnings=F, message=F, comment=NA, warning=F, results='as.is', fig.width = 10, fig.height = 6) #out.width=700, 
library(pander)
panderOptions('table.split.table', Inf)
set.seed(1)
library(dplyr)
options(stringsAsFactors = FALSE)
```

# Methods

## Survival analysis of gene expression data from TCGA

Level 3 gene expression data summarized as RSEM values was obtained using the TCGA2STAT R package v 1.2, along with the corresponding clinical annotations. Data for each of the 34 cancers was obtained separately. The data was log2-transformed and analyzed using Kaplan-Meyer curves and Cox proportional hazard model. Each gene of interest was analyzed for its effect on survival by separating patients into high/low expression subgroups. A modified approach from [@Mihaly:2013aa] was used to estimate the best gene expression cutoff that separates high/low expression subgroups with differential survival.

We took the advantage of the availability of clinical annotations. To identify if expression of a gene of interest affects survival in any specific clinical subgroup, subsets of patients annotated with specific clinical annotations were selected (e.g., “males” or “females” in the “gender” clinical annotation). Subgroups with < 40 patients were not considered.

```{r}
# Settings
gene <- "MIA" # Search and replace the name of the gene
cancer <- "BRCA"
```

# Analysis 1: `r gene` in `r cancer` cancer

```{r echo=FALSE, out.height='250px'}
knitr::include_graphics('res.genes.Analysis1/MIA_BRCA.png')
```


# Analysis 2: `r gene` in all cancers

The barplot shows the significance of `r gene` expression on survival in a given cancer. The wider (higher) the bar the more significant survival effect the gene has. See abbreviations of cancer types at [http://www.liuzlab.org/TCGA2STAT/CancerDataChecklist.pdf](http://www.liuzlab.org/TCGA2STAT/CancerDataChecklist.pdf)   

```{r echo=FALSE, out.height='550px'}
knitr::include_graphics('res.genes.Analysis2/MIA_all_TCGA_cancers.png')
```


The same data in table format. Legend:

- `Cancer`, `Cancer.Name` - cancer abbreviation and description
- `p.value` - significance of the survival effect, sorted from smallest to largest
- `HR` - hazard ratio, >1 means low expression is better for survival
- `Min.`, `Median`, `Mean`, `Max.` - log2-transformed RSEM expression level summary of the gene in a corresponding cancer. Low range of expression (e.g., Max < 6) is undesirable.
- `Cutoff_value` - gene expression cutoff best discriminating survival, found by automatic scanning. A cutoff near Min or Max is undesirable.

```{r}
cancers <- openxlsx::read.xlsx("data.TCGA/TCGA_cancers.xlsx")
mtx <- read.table("res.genes.Analysis2/global_stats.txt", sep = "\t", header = TRUE)
mtx <- mtx[order(mtx$p.value), ]
mtx <- left_join(mtx, cancers, by = c("Cancer" = "Acronym"))
mtx <- mtx[, c("Cancer", "Cancer.Name", "p.value", "HR", "Min.", "Median", "Mean", "Max.", "Cutoff_value")] # "HR_left", "HR_right", "X1st.Qu.", "X3rd.Qu.", 
# DT::datatable(mtx)
rownames(mtx) <- NULL
pander(mtx)
```

&nbsp;

# Analysis 3: `r gene` in `r cancer`, clinical subtypes

The table lists clinical subtypes where the expression of `r gene` gene in `r cancer` most significantly affects survival. The table is sorted by increasing p-values, most significant on top. Description of clinical subtypes can be found at [https://gdc.cancer.gov/about-data/data-harmonization-and-generation/clinical-data-harmonization](https://gdc.cancer.gov/about-data/data-harmonization-and-generation/clinical-data-harmonization)

```{r}
global_stats <- read.table("res.genes.Analysis3/global_stats.txt", sep = "\t", header = TRUE, fill = TRUE)
global_stats <- global_stats[order(global_stats$p.value), c("Cancer", "p.value", "HR", "Min.", "Median", "Mean", "Max.", "Cutoff_value")]
rownames(global_stats) <- NULL
pander(global_stats)
```

## Survival effect in the selected subtype

### MIA_BRCA-breast_carcinoma_estrogen_receptor_status-Positive.png

```{r echo=FALSE, out.height='240px'}
knitr::include_graphics('res.genes.Analysis3/MIA_BRCA-breast_carcinoma_estrogen_receptor_status-Positive.png')
```

### MIA_BRCA-TP53_mut_status-p53wt

```{r echo=FALSE, out.height='240px'}
knitr::include_graphics('res.genes.Analysis3/MIA_BRCA-TP53_mut_status-p53wt.png')
```


# Analysis 5: Clinical-centric analysis. Selected cancer, selected clinical subcategory, gene expression differences across categories

Expression of `r gene` in selected clinical subcategories

```{r echo=FALSE, out.height='200px'}
knitr::include_graphics('res.genes.Analysis5/BRCA_MIA_PAM50Call_RNAseq.png')
```

```{r eval=FALSE}
global_stats <- read.table("res.genes.Analysis5/global_stats.txt", sep = "\t", header = TRUE, fill = TRUE)
global_stats <- global_stats[order(global_stats$p.value), c("Cancer", "Gene", "p.value", "HR")]
rownames(global_stats) <- NULL
pander(global_stats)
```

## ANOVA and Tukey's test

What are the means of log2-expression per clinical subgroup""

```{r}
load(file = "res.genes.Analysis5/mtx_to_plot.rda")
tapply(mtx_to_plot$Gene, mtx_to_plot$Clinical, mean)
```

### ANOVA

Is the expression of gene `r gene` significantly different across clinical subgroups? Significant "Pr(>F)" suggests "Yes"

```{r}
# ANOVA
groupdiff <- lm(mtx_to_plot$Gene ~ mtx_to_plot$Clinical)
# summary(groupdiff)
anova(groupdiff)
```

### Tukey HSD (Honest Significant Difference) test

Which pair of clinical categories has significant differences? "p.adj" and confidence intervals that do not cross 0 suggest significant differences in gene expression between the subgroups in the corresponding pairwise comparison.

```{r}
# Tukey HSD
a1 <- aov(mtx_to_plot$Gene ~ mtx_to_plot$Clinical)
# summary(a1)
posthoc <- TukeyHSD(x=a1, 'mtx_to_plot$Clinical', conf.level=0.95)
posthoc %>% pander()
par(las = 2)
par(mar = c(5, 8, 5, 8))
plot(posthoc)
```


# References

